components/ee/ws-scheduler/pkg/scheduler/scheduler.go

```diff
@@ -290,22 +304,50 @@ func (s *Scheduler) schedulePod(ctx context.Context, pod *corev1.Pod) (err error
 		return nil
 	}
 
-	// we found a node for this pod - bind it to the node
-	err = s.bindPodToNode(ctx, pod, nodeName)
-	if err != nil {
-		errStr := err.Error()
-		if strings.Contains(errStr, "is already assigned to node") {
-			// some other worker was faster: This can happen and is good: do nothing
-			log.WithFields(owi).WithField("name", pod.Name).Debugf("pod already bound - someone was faster than me!")
-			return nil
+	// mark as already scheduled, even before the actually scheduling has happened.
+	// this enables asynchronous scheduling - we just need to make sure we release the binding in case the scheduling fails!
+	s.localBindingCache.markAsScheduled(pod, nodeName)
+
+	// we found a node for this pod: (asynchronously) bind it to the node! schedulingPodMap makes sure we do not try to
+	go func() {
```

Comment:
We should keep an eye on `go_goroutines` of ws-scheduler (should be on the Go in-depth dashboard).
Especially during a load test I wonder how we're doing here. We might have to introduce a simple worker pool instead of creating an unbounded amount of Go routines.

Reply:
Good point. Though due to the checks above the # of goroutines can never be higher than the # of pods that want to be schduled (~100 during load test), which I deemed fine - but I have no feeling if that correct or not.

Will look up on worker pool and overflow-mode nonetheless.

Reply:
A worker pool is easy enough to build, it's basically
```go
inc := make(chan func())
for i:=0; i<workerCount; i++ {
  go func() {
    for f := range inc {
      f()
    }
  }()
}
```

Reply:
I agree though - I don't think that's an issue for now.

Reply:
Ah, thought of it as and standard-library thing.